website/docs/feature-flag-tutorials/rust/implementing-feature-flags-rust.md

```diff
@@ -0,0 +1,286 @@
+use rgb::FromSlice;
+use std::error::Error;
+use unleash_api_client::{client, config::EnvironmentConfig};
+use webp::Encoder as WebPEncoder;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let img = ImageReader::open("input.png")?.decode()?;
+
+    // Convert to WebP
+    let webp_data = WebPEncoder::from_image(&img)?.encode(75.0);
+    std::fs::write("output.webp", webp_data.to_vec())?;
+
+    Ok(())
+}
+
+```
+
+Run the code
+
+```sh
+cargo run
+```
+
+You should see another image named `output.webp` in your folder. Make sure that the image is the same and that the compression worked correctly.
+
+![Another image named `output.webp` is now in your folder](./webp-image-in-folder.png)
+
+## 3. Add Avif support
+
+Now let's add support for Avif. We'll use a crate named `ravif` for this, which is a pure Rust converter. Later down the line, we'll rely on a boolean feature flag to toggle between the WebP and the Avif conversion.
+
+```rust
+
+use image::io::Reader as ImageReader;
+use ravif::{Encoder as AvifEncoder, Img};
+use rgb::FromSlice;
+use std::error::Error;
+use unleash_api_client::{client, config::EnvironmentConfig};
+use webp::Encoder as WebPEncoder;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let img = ImageReader::open("input.png")?.decode()?;
+
+    // Convert to WebP
+    let webp_data = WebPEncoder::from_image(&img)?.encode(75.0);
+    std::fs::write("output.webp", webp_data.to_vec())?;
+
+    // Convert to AVIF
+    let rgba_image = img.to_rgba8();
+    let pixels_rgba = rgba_image.as_raw().as_rgba();
+    let avif_data = AvifEncoder::new().with_quality(75.0).encode_rgba(Img::new(
+        pixels_rgba,
+        rgba_image.width() as usize,
+        rgba_image.height() as usize,
+    ))?;
+    std::fs::write("output.avif", avif_data.avif_file)?;
+
+    Ok(())
+}
+
+```
+
+Not as easy as the `webp` crate, I had to use the `rgb` crate to turn our image into a format that `ravif` would accept, but it works. Be mindful of the `usize` when passing the width and height.
+
+Run the code again:
+
+```sh
+cargo run
+```
+
+You should see another image named `output.avif` in your folder, alongside the `.webp` image. Make sure that all images are the same before continuing.
+
+## 4. Add Unleash to your Rust app
+
+Now, let’s connect our project to Unleash so that you can toggle that feature flag at runtime. If you wanted to, you could also do a gradual rollout, run A/B tests, etc.
+
+You’ll need 3 things:
+
+-   The Unleash SDK installed.
+-   The URL of your Unleash instance’s API. It’s `[http://localhost:4242/api/](http://localhost:4242/api/)` for your local version. You’ll want to replace it with your remote instance.
+-   The API token we created on our Unleash instance, feel free to create another one if you can’t find it.
+
+Let's first install all the SDK and all its dependencies.
+
+```sh
+cargo add enum-map@2.0.3
+cargo add reqwest --features json
+cargo add serde --features derive
+cargo add tokio --features full
```

Comment:
What version of tokio did you use? This doesn't seem to work on 1.38 which is latest in the 1.x range

I'd maybe consider using the threaded example as a base for this: https://github.com/Unleash/unleash-client-rust/blob/main/examples/threads.rs

It's conceptually simpler and doesn't need to deal with Tokio's absolute lack of commitment to stability

Reply:
Okay I have no idea what I did but this does work now...  maybe you can just ignore me on this one